report_thesis/src/sections/proposed_approach/testing_validation.tex

```diff
@@ -47,76 +61,142 @@ \subsection{Validation and Testing Procedures for Model Evaluation}\label{subsec
+\State \label{line:seed} Set random seed for reproducibility if $\text{seed}$ is not None
+\State \label{line:remove_duplicates} Remove duplicate entries based on $g$ and sort by $t$
+\State \label{line:assign_folds} Assign fold numbers sequentially from 0 to $k-1$ to unique targets
 \If{extreme values handling is enabled}
-    \State \label{line:identify_extremes} Identify extreme values at percentiles \( p \) and \( 1-p \)
-    \State \label{line:reassign_extremes} Reassign extreme values to folds \( 0 \) to \( k-2 \)
+    \State \label{line:identify_extremes} Identify extreme values at percentiles $p$ and $1-p$
+    \State \label{line:reassign_extremes} Reassign extreme values to folds $0$ to $k-2$
 \EndIf
-\State \label{line:merge_folds} Merge fold assignments back into the original dataset
-\State \label{line:split_dataset} Split dataset into test set \( \mathbf{D}_\text{test} \) (fold \( k-1 \)) and remaining data \( \mathbf{D}_\text{train} \)
-\State \label{line:create_folds} Create \( k-1 \) training and validation folds
-\For{each fold \( i \) from 0 to \( k-2 \)}
-    \State \( \mathbf{F}_\text{train}[i] \gets \mathbf{D}_\text{train} \setminus \text{fold}_i \)
-    \State \( \mathbf{F}_\text{val}[i] \gets \text{fold}_i \)
-    \State Append \((\mathbf{F}_\text{train}[i], \mathbf{F}_\text{val}[i])\) to \(\mathbf{F}_\text{cv}\)
+\State \label{line:merge_folds} Merge fold assignments information into the original dataset
+\State \label{line:split_dataset} Split dataset into test set $\mathbf{D}_\text{test}$ (fold $k-1$) and remaining data $\mathbf{D}_\text{train}$
+\State \label{line:create_folds} Create $k-1$ training and validation sets
+\For{each fold $i$ from 0 to $k-2$}
+    \State $\mathbf{T}_\text{train}[i] \gets \mathbf{D}_\text{train} \setminus \text{fold}_i$
+    \State $\mathbf{T}_\text{val}[i] \gets \text{fold}_i$
+    \State Append $(\mathbf{T}_\text{train}[i], \mathbf{T}_\text{val}[i])$ to $\mathbf{T}_\text{cv}$
 \EndFor
 \State \label{line:remove_fold_column} Remove fold column from all datasets
-\State \Return \( \mathbf{F}_\text{cv}, \mathbf{D}_\text{train}, \mathbf{D}_\text{test} \)
+\State \Return $\mathbf{T}_\text{cv}, \mathbf{D}_\text{train}, \mathbf{D}_\text{test}$
 \end{algorithmic}
 \end{algorithm}
 
 The procedure outlined in Algorithm~\ref{alg:custom_kfold_cv} begins by setting a random seed for reproducibility if one is provided (Line~\ref{line:seed}).
 This ensures that the results are consistent across different runs of the algorithm.
-Next, the dataset is processed to remove any duplicate entries based on the group column and then sorted by the target column (Line~\ref{line:remove_duplicates}).
+Next, the dataset is processed to remove any duplicate entries based on the group column $g$ and then sorted by the target column $t$ (Line~\ref{line:remove_duplicates}).
```

Comment:
Added these because it can be hard to distinguish between when we're referencing the dataset and variables in the algorithm